### PR TITLE
Transform URL services to other variables

### PR DESCRIPTION
**Pull request by sroze**

### CODE REVIEW

File: php/shared/usr/local/share/php/common_functions.sh [hunk 1]
```diff
@@ -60,3 +60,33 @@ composer_package_compare() {
...
+  hostport=`echo $url | sed -e s,$userpass@,,g | cut -d/ -f1`
+  port=`echo $hostport | grep : | cut -d: -f2`
+  if [ -n "$port" ]; then
+    host=`echo $hostport | grep : | cut -d: -f1`
+  else
+    host=$hostport
+  fi
+
+  # extract the path (if any)
+  path="`echo $url | grep / | cut -d/ -f2-`"
```
Comment (andytson-inviqa) [inline php/shared/usr/local/share/php/common_functions.sh]: I'd still like to see the php parseurl used, just as it's one less thing that could be wrong.

Also Kieran is thinking about using some concurency, which could cause issues with these global proto/scheme/user/password/path will have issues with unless passed as a sub-shell

Comment (andytson-inviqa) [reply php/shared/usr/local/share/php/common_functions.sh]: not that I know a good way to use parseurl and collect the arguments though

Comment (sroze) [reply php/shared/usr/local/share/php/common_functions.sh]: OK, that's fair enough. Then... how would the result of this function look like then? /cc @kierenevans

**Comment by sroze** [comment]
@kierenevans @dcole-inviqa @andytson-inviqa I know it's used by a few projects. Do you want to merge or take it over (i.e. do the change to use PHP instead of bash) ?

**Comment by sroze** [comment]
I'll close it then.

**Pull request closed without merging**